dip-aj-hd-mixing.md

```diff
@@ -0,0 +1,152 @@
+keywords: CoinJoin PrivateSend mobile coin mixing mix hdpath mixpath
+
+## Prior Art
+
+- [Multi-Account Hierarchy for Deterministic Wallets](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)
+
+## Motivation
+
+Currently there's no strategy for tracking "coin mixing" on mobile devices, or
+sharing "coin mixing" enabled wallet accounts between devices - such as in the
+case where a second device should do the mixing because the mobile device cannot
+reasonably do so.
+
+Preferably we want a strategy that doesn't require syncing indexes, especially
+not in one-off formats.
+
+This solves all of those problems with minimal complexity.
+
+This won't break any existing wallets - those that are unaware of mixing will
+still be aware - and will enable shared data via an offline indexing scheme
+protocol rather than requiring a new or custom format or online synced data.
+
+## Specification
+
+[lextx]: https://github.com/bitcoin/bips/blob/master/bip-0069.mediawiki
+
+[Lexicographical Indexing of Transaction Inputs and Outputs][lextx] MUST be used
+for all mixing transactions.
+
+The _Usage_ component (at index 4) of the HDPath is designated for tracking coin
+mixes for mixed accounts.
+
+HDPaths are used in this format:
+
+```text
+m/<bip44>'/<coin>'/<account>'/<mixcount>/<index>
+m/44'/5'/0'/0/0
+```
+
+Accounts should be designated as either Mixed or Unmixed. However, if any path
+in the format of `m/44'/5'/<account>'/0/<index>` has only a single coin which is
+a recognizable denomination, it should be searched from
+`m/44'/5'/<account>'/1/<index>` up through `m/44'/5'/<account>'/<n>/<index>` to
+the first unused address to find unspent mixed, denominated coins.
+
+### Coin Mix Tracking via HDPath
+
+**Summary of BIP-44**
+
+The index to a particular private key or address of an HDPath is defined in this
+format:
+
+```text
+m/<bip44>'/<coin>'/<account>'/<usage>/<index>
+m/44'/5'/0'/0/0
+```
+
+The _Usage_ path component is presently used to indicate either:
+
+- `0` as _External_, for sending money
+- `1` as _Internal_ for receiving change
+
+**Applied to Coin Mixing**
+
+There are many factors that effect the viability of coin mixing, so for this
+specification and these examples we'll make 2 assumptions:
+
+1. mixed coins are generally not used in a scenarios where change is returned
+2. any return change is denominated and its transaction adheres to
+   [Lexicographical Indexing of TX Inputs & Outputs][lextx]
+
+Given those assumptions either A) _Usage_ will _never_ represent return change
+or B) the return change will already have some measure of entropy added to it.
+
+This means that the _Usage_ path can be used in for tracking the mix count with
+coming into conflict with unmixed coins.
+
+```text
+m/<bip44>'/<coin>'/<account>'/<mixcount>/<index>
+m/44'/5'/0'/0/0 - newly denominated coin
+m/44'/5'/0'/1/0 - coin which has been mixed once
+m/44'/5'/0'/...
+m/44'/5'/0'/16/0 - a fully mixed coin
+m/44'/5'/0'/... - an overmixed coin
```

Comment:
Currently, the DIP9 format is being used for CoinJoin development on mobile, which is based on using number 4 from
https://github.com/dashpay/dips/blob/master/dip-0009/assignments.md.

For example with mainnet, the path is the following:
`m/9'/5'/4'/x`

This follows the following scheme.
`m / 9' / coin_type' / feature' / *`

Presently, what is used doesn't include the account number.  This will need to be added in the mobile scheme.